ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -1130,21 +1129,17 @@ impl Module {
 }
 
 /// Deserialise an AOT module from the slice `data`.
-pub(crate) fn deserialise_module(data: &[u8]) -> Result<Module, CompilationError> {
-    match Module::from_bytes((data, 0)) {
-        Ok(((_, _), mut modu)) => {
-            modu.compute_local_operand_func_indices();
-            Ok(modu)
-        }
-        Err(e) => Err(CompilationError::Unrecoverable(e.to_string())),
-    }
+pub(crate) fn deserialise_module(data: &[u8]) -> Result<Module, Box<dyn Error>> {
```

Comment:
In the description you said:

> This PR starts, crudely, to split apart these two classes of errors. It tries to make it so that invariant-breaking causes the system to die at the first point that we notice things going wrong

I'm therefore not sure I understand this PR. Why don't you just panic here instead of bubbling an error up?

Reply:
It's a minor design choice in this particular regard IMHO.

Reply:
I don't understand. You said you want to crash ASAP for cases where the system can't operate, right? Here is one such case, but you opted to bubble up an error. So have I misunderstood something?

Reply:
In this case it just goes up one level. Personally I would inline this function, but I suggest not overthinking it.